README.md

```diff
@@ -132,6 +170,50 @@ automatically even if the code inside the `do` scope throws an error.)
 Conversely, `load(::Stream)` and `save(::Stream)` should not close the
 input stream.
 
+`loadstreaming` and `savestreaming` use the same query mechanism, but return a
+decoded stream that users can `read` or `write`. You should also implement a
+`close` method on your reader or writer type. Just like with `load` and `save`,
+if the user provided a filename, your `close` method should be responsible for
+closing any streams you opened in order to read or write the file. If you are
+given a `Stream`, your `close` method should only do the clean up for your
+reader or writer type, not close the stream.
+
+```jl
+struct WAVReader
+    io::IO
+    ownstream::Bool
+end
+
+function read(reader::WAVReader, frames::Int)
+    # read and decode audio samples from reader.io
+end
+
+function close(reader::WAVReader)
+    # do whatever cleanup the reader needs
+    reader.ownstream && close(reader.io)
+end
+loadstreaming(f::File{format"WAV"}) = WAVReader(open(f), true)
```

Comment:
Add blank line before this one? Or does the "fallback" comment below also apply to `close`?

Reply:
re-ordered the comment to clarify this